sw/device/lib/dif/dif_sensor_ctrl.h

```diff
@@ -12,13 +12,129 @@
  */
 
 #include <stdint.h>
-
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_base.h"
 #include "sw/device/lib/dif/autogen/dif_sensor_ctrl_autogen.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif  // __cplusplus
 
+/**
+ * An event, identified by a numeric id.
+ *
+ */
+typedef uint32_t dif_sensor_ctrl_event_idx_t;
+
+/**
+ * A vector of AST events, where each bit represents one event.
+ *
+ */
+typedef uint32_t dif_sensor_ctrl_events_t;
+
+/**
+ * IO power status
+ *
+ */
+typedef uint32_t dif_sensor_ctrl_io_power_status_t;
+
+/**
+ * Locks sensor control configuration from further updates.
+ *
+ *
+ * @param sensor_ctrl A sensor_ctrl handle.
+ * @return 'kDifBadArg' if `handle` is null.
+ */
+OT_WARN_UNUSED_RESULT
+dif_result_t dif_sensor_ctrl_lock_cfg(const dif_sensor_ctrl_t *sensor_ctrl);
+
+/**
+ * Sets the value for a partciular AST event trigger.
+ *
+ * @param sensor_ctrl A sensor_ctrl handle.
+ * @return 'kDifBadArg' if `handle` is null or if `event_idx` is larger than the
+ * number of events supported.
+ */
+OT_WARN_UNUSED_RESULT
+dif_result_t dif_sensor_ctrl_set_ast_event_trigger(
+    const dif_sensor_ctrl_t *sensor_ctrl,
+    dif_sensor_ctrl_event_idx_t event_idx, dif_toggle_t enable);
+
+/**
+ * Sets the fatality configuration of a particular event.
+ *
+ * An event can be configured to be either fatal or recoverable.
+ *
+ * @param sensor_ctrl A sensor_ctrl handle.
+ * @param en_fatal The fatality enablement state of an event.
```

Comment:
Consider: 

```suggestion
 * @param en_fatal The fatality enablement state of an event. If
 * `kDifToggleEnabled`, the event is fatal. Otherwise, it is recoverable.
```

Reply:
added